pkg/sensors/config/confmap/confmap.go

```diff
@@ -120,11 +124,13 @@ func UpdateTgRuntimeConf(mapDir string, nspid int) error {
 
 	log.WithFields(logrus.Fields{
 		"confmap-update":                configMap.Name,
-		"deployment.mode":               cgroups.DeploymentCode(deployMode).String(),
-		"cgroup.fs.magic":               cgroups.CgroupFsMagicStr(cgroupFsMagic),
+		"deployment.mode":               cgroups.DeploymentCode(v.Mode).String(),
```

Comment:
I've got this far but still don't really understand how to set this deployment.mode thing. :/

Reply:
> I've got this far but still don't really understand how to set this deployment.mode thing. :/

It is automatically set, you don't have to do anything, when we start the observer we auto detect all this:

https://github.com/cilium/tetragon/blob/main/pkg/cgroups/cgroups.go#L612

That will trigger findMigrationPath() if it was not run before: https://github.com/cilium/tetragon/blob/92a2503f7788e3707f61c99cd2751667f9b326cc/pkg/cgroups/cgroups.go#L496

That calls Detect Cgroup Mode: https://github.com/cilium/tetragon/blob/92a2503f7788e3707f61c99cd2751667f9b326cc/pkg/cgroups/cgroups.go#L507

Then parse own cgroup paths according to mode, we do separate the hybrid from unified cgroup mode, cause if in hybrid mode systemd will use the cgroupv2 hierarchy only for process tracking where k8s will use legacy cgroupv1 for proper resources tracking. So back we parse the results paths here:

cgroupv1: https://github.com/cilium/tetragon/blob/92a2503f7788e3707f61c99cd2751667f9b326cc/pkg/cgroups/cgroups.go#L324

cgroupv2: https://github.com/cilium/tetragon/blob/92a2503f7788e3707f61c99cd2751667f9b326cc/pkg/cgroups/cgroups.go#L424

Where both will try to detect if we are cgroup namespaced etc, and finally call:
setDeploymentMode() on the final constructed cgroup path of Tetragon https://github.com/cilium/tetragon/blob/92a2503f7788e3707f61c99cd2751667f9b326cc/pkg/cgroups/cgroups.go#L246 

Of course this part of the code can be improved but don't want to touch it until we advance with e2e tests then easy to improve...

Reply:
This was removed from this PR, and it will be re-introduced in follow up one.